numpy/core/src/multiarray/mapping.c

```diff
@@ -1125,7 +1125,7 @@ array_boolean_subscript(PyArrayObject *self,
         ret = (PyArrayObject *)PyArray_NewFromDescrAndBase(
                 Py_TYPE(self), dtype,
                 1, &size, PyArray_STRIDES(ret), PyArray_BYTES(ret),
-                PyArray_FLAGS(self), (PyObject *)self, (PyObject *)self);
+                PyArray_FLAGS(self), (PyObject *)self, (PyObject *)tmp);
```

Comment:
Needs a decref of `tmp` too after this line. Caused by the removal of [this line](https://github.com/numpy/numpy/pull/11246/files#diff-6c6d7467bec90e023c600b31f151a5b1L1139), which previously stole a reference, but was replaced by borrowing one.